File: LayoutTests/http/tests/contentextensions/resources/save-ping-and-redirect-to-save-ping.php
<?php
require_once 'ping-file-path.php';

$DO_NOT_CLEAR_COOKIES = true;
require_once 'save-ping.php';

header('HTTP/1.1 307');
header('Location: save-ping.php' . (isset($_SERVER['QUERY_STRING']) ? '?' . $_SERVER['QUERY_STRING'] : ''));
?>

File: LayoutTests/http/tests/contentextensions/resources/save-ping-and-redirect-to-save-ping.py
#!/usr/bin/env python3

import os
import sys
from save_ping import save_ping

query_string = os.environ.get('QUERY_STRING', None)
if query_string is not None:
    query_string = '?{}'.format(query_string)

save_ping(True)

sys.stdout.write(
    'status: 307\r\n'
    'Location: save-ping.py{}\r\n'
    'Content-Type: text/html\r\n\r\n'.format(query_string)
)